File: VNC.CodeAnalysis Test Cases/TestCasesVB/Quality/GoToLabels.vb
Public Class GoToLabels
    Public Shared Sub message()
    End Sub

    Public Sub gotoFun()
        Dim array As Integer(,) = Nothing
        For i As Integer = 0 To 2 - 1
            For j As Integer = 0 To 9 - 1
                Dim flag As Boolean = array(i, j).Equals(3)
                If flag Then
                    GoTo IL_41
                End If
            Next
        Next
IL_41:
        GoToLabels.chump()
    End Sub

    Private Shared Sub chump()
        Dim i As Integer = 0
        Dim cost As Integer = 0
        Select Case i
            Case 1
            Case 2
                cost += 25
            Case 3
                cost += 50
            Case Else
                Return
        End Select
        cost += 25
    End Sub

End Class

File: CodeAnalysis Test Cases/TestCasesCS/Quality/GoToLabels.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace CSTestCases.Quality
{
    public class GoToLabels
    {
        public static void message()
        {
        }
        public void gotoFun()
        {
            // Search:
            Int32[,] array = null;

            for (int i = 0; i < 2; i++)
            {
                for (int j = 0; j < 9; j++)
                {
                    if (array[i, j].Equals(3))
                    {
                        goto Found;
                    }
                }
            }

            Found:
            chump();
        }

        static void chump()
        {
            Int32 n = 0;

            int cost = 0;
            switch (n)
            {
                case 1:
                    cost += 25;
                    break;
                case 2:
                    cost += 25;
                    goto case 1;
                case 3:
                    cost += 50;
                    goto case 1;
                default:
                    break;
            }
        }
    }
}
